opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -336,6 +335,10 @@ def _filter_attribute_values(attributes: types.Attributes):
                 attributes.pop(attr_key)
 
 
+def make_immutable_dict(attributes):
```

Comment:
Nit: We should make this protected (underscore). Also, for the naming, this implies that it can be used for any mapping type, so I would either change the parameter name, or change the function name to `_create_immutable_attributes` or something like that.

Reply:
Great feedback again, thank you.